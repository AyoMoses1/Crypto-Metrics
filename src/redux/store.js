import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './cryptos/cryptoSlice';

const store = configureStore({
  reducer: {
    cryptoReducer,
  },
});

export default store;
