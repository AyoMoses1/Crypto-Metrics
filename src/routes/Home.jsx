import React from 'react';
import { useSelector } from 'react-redux';

function Home() {
  const state = useSelector((state) => state.cryptoReducer.list);
  return (
    <>
      {state.map((coin) => (
        <div key={coin.id}>
          <div>{coin.name}</div>
          <div className="image">
            <img src={coin.image} alt={coin.id} />
          </div>
        </div>
      ))}
    </>
  );
}

export default Home;
