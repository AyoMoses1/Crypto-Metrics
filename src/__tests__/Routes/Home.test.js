import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Home from '../../routes/Home';

it('App component test', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
