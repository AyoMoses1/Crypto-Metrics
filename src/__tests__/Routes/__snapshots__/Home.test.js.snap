// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test 1`] = `
<div
  className="home-wrapper container"
>
  <div
    className="banner"
  >
    <h2>
      Covid Tracker
    </h2>
    <p>
      Over 200views
    </p>
  </div>
  <div
    className="grid-container justify-content-center"
  />
</div>
`;
